Sage/carbon
# Cypress accessibility box carbon provider

### Proposed behaviour
#### Box
- Add `accessibility` Cypress tests for the `Box` component to use the new cypress-component-react framework for testing.
- Move component stories to the `box.stories.tsx`
- Refactor for `box.stories.mdx` file
- Move testing component to the `box-test.stories.tsx`

#### Carbon Provider
- Add `accessibility` Cypress tests for the `Carbon Provider` component to use the new cypress-component-react framework for testing.
- Move component stories to the `carbon-provider.stories.tsx`
- Refactor for `carbon-provider.stories.mdx` file

### Current behaviour
Old approach for accessibility tests

### Checklist



#### QA


### Additional context



### Testing instructions

- [x] Run `npx cypress open --component` to check if there are newly added tests
- [x] Check if the `box.test.js` file passed
- [x] Check if the `carbon-provider.test.js` file passed
- [x] Run `npx cypress run --component` to check none of the other *.test.js files have regressed
- [x] Run `npx cypress run --e2e` to check none of the feature files have regressed
Run `npm run build-storybook` -> `npx sb extract` -> `npx http-server storybook-static -p 9001` and run `npx cypress run --e2e` and check that:
- [x] `Box` tests are not running twice.
- [x] `Carbon Provider` tests are not running twice.
